matlab_simulation/05-estimation/fastSLAM/ekf_correct.m

```diff
@@ -0,0 +1,25 @@
+%% A function for the correction step of the EKF
+% NOTE: THE prediction step has already been called, Meas Jac Calculated
+% and new feature initialized, if at all. This is the next correction step
+% 
+% INPUTS:(in order of call)
+% -------
+% Mean of all observed Features
+% covariance of all Observed Features
+% Jacobian of the Measurement with respect to the features
+% Predicted Observation from the prediction step of EKF
+% Measurement Uncertainty
+% OUTPUTS:(in order of call)
+% --------
+% Preicted mean of The feature
+% Predicted covariance of the feature 
+% % AUTHOR: BISMAYA SAHOO, EMAIL:bsahoo@uwaterloo.ca
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+function [meanFeatP,covaFeatP]=ekf_correct(muFeat,covFeat,obsFeat,H_feat,pred_obs,measUncertainty)
+        I = obsFeat(:,1)-pred_obs;
```

Comment:
No. If you look at the way the ekf_correct function is called, it loops through all of the observed features and processes them one by one.